### PR TITLE
Added documentation for the plugins.query.field_type_tolerance setting (#1300)

### LINKED ISSUE #1300: [DISCUSSION] Properly support array values in new engine
## What is the bug?
The new engine does not return values in an array while the Legacy engine returns all values in a row as an array. Implementing same support as `V1` does isn't a right way, because legacy engine produces inconsistent value.

## How can one reproduce the bug?
1. Start OpenSearch server
2. Clean index if was created before: `curl -XDELETE 'http://localhost:9200/dbg'`
3. Create a simple index with automatic mapping: `curl -X POST "localhost:9200/dbg/_doc/?pretty" -H 'Content-Type: application/json' -d '{"myNum": 5}'`
4. Query data: `select * from dbg`. Not bas so far.
5. Add new doc: `curl -X POST "localhost:9200/dbg/_doc/?pretty" -H 'Content-Type: application/json' -d '{"myNum": [3, 4]}'`
6. Check mapping: `curl -X GET "localhost:9200/dbg?pretty"`
```json
"mappings" : {
  "properties" : {
	"myNum" : {
	  "type" : "long"
	}
  }
}
```
7. Query in the new engine: `curl -s -XPOST http://localhost:9200/_plugins/_sql -H 'Content-Type: application/json' -d '{"query": "select * from dbg"}'`
```json
{
  "schema": [
    {
      "name": "myNum",
      "type": "long"
    }
  ],
  "datarows": [
    [
      5
    ],
    [
      3
    ]
  ],
  "total": 2,
  "size": 2,
  "status": 200
}
```
(if you have only second doc in the index)
```json
"schema": [
    {
      "name": "myNum",
      "type": "long"
    }
  ],
  "datarows": [
    [
      3
    ],
    [
      4
    ]
  ]
```
8. Query in the legacy engine: `curl -s -XPOST http://localhost:9200/_plugins/_sql -H 'Content-Type: application/json' -d '{"query": "select * from dbg", "fetch_size": 20}'`
```json
{
  "schema": [
    {
      "name": "myNum",
      "type": "long"
    }
  ],
  "total": 2,
  "datarows": [
    [
      5
    ],
    [
      [
        3,
        4
      ]
    ]
  ],
  "size": 2,
  "status": 200
}
```

## What is the expected behavior?
**TBD**

### Why legacy response is incorrect?
It declares data type as long, but returns a number **and** array of numbers. Imagine a user has a parser for response, what should parser do with such values?
You can try our JDBC driver as an example of a customer application.

## What is your host/environment?
`main` @ 6108ca1a1

### PR DESCRIPTION
**Pull request by normanj-bitquill**
### Description
Add documentation for the `plugins.query.field_type_tolerance` setting.

### Related Issues
Resolves #1300

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
 - [ ] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/user/admin/settings.rst [hunk 1]
```diff
@@ -824,3 +824,55 @@ To Re-enable Data Sources:::
       }
     }
 
+plugins.query.field_type_tolerance
```
Comment (penghuo) [inline docs/user/admin/settings.rst:827]: Add to `Table of contents`

File: docs/user/admin/settings.rst [hunk 2]
```diff
@@ -824,3 +824,55 @@ To Re-enable Data Sources:::
...
+==================================
+
+Description
+-----------
+
+This setting controls whether preserve arrays. If this setting is set to false, then an array is reduced
```
Comment (penghuo) [inline docs/user/admin/settings.rst:833]: highlight default value is True.

File: docs/user/admin/settings.rst [hunk 3]
```diff
@@ -824,3 +824,55 @@ To Re-enable Data Sources:::
...
+
+    {"name": "one", "value": 1}
+
+1. The default value is true (preserve arrays)
+2. This setting is node scope
+3. This setting can be updated dynamically
+
+Update Settings Request::
+
+    sh$ curl -sS -H 'Content-Type: application/json' -X PUT 'localhost:9200/_cluster/settings?pretty' \
```
Comment (penghuo) [inline docs/user/admin/settings.rst:866]: could u follow plugins.query.size_limit setting doc style.

File: docs/user/admin/settings.rst [hunk 4]
```diff
@@ -824,3 +824,55 @@ To Re-enable Data Sources:::
...
+plugins.query.field_type_tolerance
+==================================
+
+Description
+-----------
+
+This setting controls whether preserve arrays. If this setting is set to false, then an array is reduced
+to the first non array value of any level of nesting.
+
+If you have an index with the following value for a field::
```
Comment (penghuo) [inline docs/user/admin/settings.rst:836]: doc is exectuable, you can run ppl/sql query directly. for example, https://github.com/opensearch-project/sql/blob/5dafb2fc5960e0a1edff70a089a5390519fc909c/docs/user/ppl/cmd/eval.rst

**Review by penghuo — COMMENTED** [review]
could u add limitation of this setting. for instance, expression will failed.

File: docs/user/admin/settings.rst (hunk 1 above)
Comment (normanj-bitquill) [reply docs/user/admin/settings.rst:827]: I see `plugins.query.field_type_tolerance` in the table of contents.
https://github.com/Bit-Quill/opensearch-project-sql/blob/issue-1300-documentation/docs/user/admin/settings.rst

File: docs/user/admin/settings.rst (hunk 2 above)
Comment (normanj-bitquill) [reply docs/user/admin/settings.rst:833]: Reworked the documentation. It should be more clear what the default is.

File: docs/user/admin/settings.rst (hunk 3 above)
Comment (normanj-bitquill) [reply docs/user/admin/settings.rst]: Used that style for the commands to update the setting.

File: docs/user/admin/settings.rst (hunk 4 above)
Comment (normanj-bitquill) [reply docs/user/admin/settings.rst]: I have reworked the documentation. It still does not actually run the queries, although they are run for other files such as `user/beyond/partiql.rst`.

**Comment by normanj-bitquill** [comment]
> could u add limitation of this setting. for instance, expression will failed.

I added a limitations section.

File: docs/user/admin/settings.rst [hunk 5]
```diff
@@ -824,3 +824,58 @@ To Re-enable Data Sources:::
...
+	      "plugins.query.field_type_tolerance" : true
+	    }
+	  }'
+
+Limitations:
+------------
+* Fields with array values should only be used in the projection list
+* Array values are not supported by SQL or PPL functions
+* Array values in expressions will cause the query to fail or produce incorrect results
+* PPL commands do not support examining or altering array values
```
Comment (penghuo) [inline docs/user/admin/settings.rst:881]: 1. Could you rephrase limitation? for instance,
```
OpenSearch does not natively support the ARRAY data type but does allow multi-value fields implicitly. The SQL/PPL plugin adheres strictly to the data type semantics defined in index mappings. When parsing OpenSearch responses, it expects data to match the declared type and does not account for data in array format. If the plugins.query.field_type_tolerance setting is enabled, the SQL/PPL plugin will handle array datasets by returning scalar data types, allowing basic queries (e.g., SELECT * FROM tbl WHERE condition). However, using multi-value fields in expressions or functions will result in exceptions. If this setting is disabled or absent, only the first element of an array is returned, preserving the default behavior.
```
2. Could u add an example of exception example and workable example in limitation page?

File: docs/user/admin/settings.rst [hunk 6]
```diff
@@ -824,3 +824,58 @@ To Re-enable Data Sources:::
...
+
+Reenable field type tolerance::
+
+    >> curl -H 'Content-Type: application/json' -X PUT localhost:9200/_plugins/_query/settings -d '{
+	    "transient" : {
+	      "plugins.query.field_type_tolerance" : true
+	    }
+	  }'
+
+Limitations:
```
[the remark this answers is not in the document]
Comment (normanj-bitquill) [reply docs/user/admin/settings.rst:876]: Added a section at the end of the `limitations.rst` file.

[the remark this answers is not in the document]
Comment (normanj-bitquill) [reply docs/user/admin/settings.rst:876]: Copied.

**Review by penghuo — APPROVED** [review]
Thx!

**Review by dai-chen — APPROVED** [review]
